File: Siddhi mam Tasks/Day 29/Broadcast_Receiver_1/app/src/main/java/com/example/broadcast_receiver_1/MainActivity.java
package com.example.broadcast_receiver_1;


import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.IntentFilter;
import android.media.AudioManager;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    AirplaneModeChangeReceiver airplaneModeChangeReceiver = new AirplaneModeChangeReceiver();

    MuteChangeReceiver muteChangeReceiver = new MuteChangeReceiver();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    @Override
    protected void onStart() {
        super.onStart();
        IntentFilter filter = new IntentFilter(Intent.ACTION_AIRPLANE_MODE_CHANGED);
        registerReceiver(airplaneModeChangeReceiver,filter);

    }

    @Override
    protected void onStop() {
        super.onStop();
        unregisterReceiver(airplaneModeChangeReceiver);

    }

    protected void onResume() {
        super.onResume();
        IntentFilter filter = new IntentFilter(AudioManager.RINGER_MODE_CHANGED_ACTION);
        registerReceiver(muteChangeReceiver, filter);
    }

    @Override
    protected void onPause() {
        super.onPause();
        unregisterReceiver(muteChangeReceiver);
    }
}

File: Siddhi mam Tasks/Day 37/Co-routines/Coroutine_hierarchy/app/src/main/java/com/example/co_routine_hierarchy/MainActivity.kt
package com.example.co_routine_hierarchy

import android.content.ContentValues.TAG
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        GlobalScope.launch(Dispatchers.Main) {
            execute()
        }



    }

    private suspend fun execute()
    {
        val parentJob = GlobalScope.launch(Dispatchers.Main) {
            Log.d(TAG,"parents started ")

            val childJob = launch (Dispatchers.IO){
                Log.d(TAG,"child job started ")
                delay(5000)
                Log.d(TAG,"child job ended ")
            }

            delay(3000)
            Log.d(TAG,"parents ended ")
        }

        parentJob.join()  // so here we can see that until our child co-routine is not completed parents completed will not print because we have use the join method

        Log.d(TAG,"parents completed")
    }
}